### PR TITLE
haunt: fix Guile load paths

### PR DESCRIPTION
**Pull request by AluisioASG**
###### Motivation for this change

The current `haunt` package doesn't actually run due to looking in the wrong paths – Guile packages may or may not include the Guile version in their install paths, and Haunt does include it.

Also include checks to catch this in the future.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/haunt/default.nix [hunk 1]
```diff
@@ -27,10 +27,22 @@ stdenv.mkDerivation rec {
     guile-reader
   ];
 
-  postInstall = ''
-    wrapProgram $out/bin/haunt \
-      --prefix GUILE_LOAD_PATH : "$out/share/guile/site:${guile-commonmark}/share/guile/site:${guile-reader}/share/guile/site" \
```
Comment (AndersonTorres) [inline pkgs/applications/misc/haunt/default.nix:32]: Why did you wipe out guile-commonmark and guile-reader here?

File: pkgs/applications/misc/haunt/default.nix [hunk 2]
```diff
@@ -27,10 +27,22 @@ stdenv.mkDerivation rec {
...
-      --prefix GUILE_LOAD_COMPILED_PATH : "$out/share/guile/site:${guile-commonmark}/share/guile/site:${guile-reader}/share/guile/site"
+  doCheck = true;
+
+  postInstall =
+    let guileVersion = lib.versions.majorMinor guile.version;
+    in
```
Comment (AndersonTorres) [inline pkgs/applications/misc/haunt/default.nix:34]:
```suggestion
    let 
      guileVersion = lib.versions.majorMinor guile.version;
    in
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/haunt/default.nix (hunk 1 above)
Comment (AluisioASG) [reply pkgs/applications/misc/haunt/default.nix:32]: The build inputs are added to the load paths by [Guile's setup hook](https://github.com/NixOS/nixpkgs/blob/bd64a6221f18a24c79b879c0441f4420f0f2d47e/pkgs/development/interpreters/guile/setup-hook-2.2.sh), so we can just prepend Haunt's path to those.

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/applications/misc/haunt/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/applications/misc/haunt/default.nix:32]: Oh, good! I will try the same in skribilo then!

**Review by AndersonTorres — COMMENTED** [review]
LGTM

**Pull request merged**